Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -735,14 +756,17 @@ protected void HandleSortColumn( DataGridColumn<TItem> column, bool changeSortDi
                 }
 
                 if ( changeSortDirection )
-                    column.CurrentSortDirection = column.CurrentSortDirection.NextDirection();
+                    column.CurrentSortDirection = sortDirection ?? column.CurrentSortDirection.NextDirection();
 
                 if ( !SortByColumns.Any( c => c.Field == column.Field ) )
                 {
                     SortByColumns.Add( column );
                 }
                 else if ( column.CurrentSortDirection == SortDirection.None )
                     SortByColumns.Remove( column );
+
+                if ( changeSortDirection )
+                    InvokeAsync( () => SortChanged.InvokeAsync( new DataGridSortChangedEventArgs( column.Field, column.CurrentSortDirection ) ) );

```

Comment:
I kinda forgot... Is the EventCallback safe to call without null check / delegate exists check?

Reply:
SortChanged is of type EventCallback (struct) soi it can never be null. It is safe.